libopenage/datastructure/pairing_heap.h

```diff
@@ -631,14 +629,10 @@ class PairingHeap final {
 		}
 	}
 
-protected:
 	compare cmp;
 	size_t node_count;
 	element_t root_node;
-
-#if OPENAGE_PAIRINGHEAP_DEBUG
 	std::unordered_set<element_t> nodes;
-#endif
```

Comment:
I wonder if we could keep this set only when `OPENAGE_PAIRINGHEAP_DEBUG` is true, so that the eap becomes even more efficient. Looks like it's only necessary for clearing up the nodes on deletion right now. I suggest we could try this solution:

1. From `clear()`, use `iter_all` to iterate through all the nodes and call `delete` on them.
2. Wrap all interactions of `nodes` with the `#if OPENAGE_PAIRINGHEAP_DEBUG` preprocessor thing again.

Reply:
alright sound like a plan :)

Reply:
@heinezen  requested changes Implemented at #1713